dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/MessageTaskTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.oracle;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.connector.oracle.OracleMetadataConnector.GroupTask;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractTask;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class MessageTaskTest {
+
+  @Test
+  public void getMessages_success() {
+    ImmutableList<String> messages =
+        ImmutableList.of(
+            "All the select tasks failed:",
+            "(1): fake : RuntimeException: ExceptionCauseA",
+            "(2): fake : RuntimeException: ExceptionCauseB");
+    MessageTask task = createTask();
+    // Act
+    ImmutableList<String> taskMessages = ImmutableList.copyOf(task.getMessages());
```

Comment:
I would suggest moving `ImmutableList.copyOf` invocation to the Assert step, in order to emphasize the return type of the `getMessage` method (the method under test) and to keep the Act step as small as possible.

Reply:
Done